.github/workflows/build.yml

```diff
@@ -7,15 +7,23 @@ on:
       - '**.rs'
       - 'Cargo.*'
       - '*/Cargo.*'
+      - '.github/workflows/*'
+  push:
+    branches:
+      - main
```

Comment:
putting it here ( instead of separate file ), will trigger this bulky workflow after merging PRs as well ( and as we are running it before merge as well, is it worth it? )

directly pushing to main branch is bad and must not be performed.

wdyt?

Reply:
> directly pushing to main branch is bad and must not be performed.

`push: branches: [main]` serves two purposes.

1. It executes when someone pushes directly to `main`, which we don't want anyway.
2. Also, it will be executed when the pull request gets merged into main, and becomes a commit on the `main` branch.

We need that to satisfy the second condition only. If you want to be extra safe, disable `push to main` from repo settings.

Reply:
> it will be executed when the pull request gets merged into main
>

workflow must pass before merging to main, it will basically be just two runs of same thing. that i why i asked

Reply:
>  just two runs of same thing

We need a separate run on the main branch to get the coverage badge from coveralls.

Reply:
what is a badge?

Reply:
in https://coveralls.io/github/bytebeamio/rumqtt:

[![Coverage Status](https://coveralls.io/repos/github/bytebeamio/rumqtt/badge.svg?branch=main)](https://coveralls.io/github/bytebeamio/rumqtt?branch=main)

Reply:
what is the use of badge? to showcase in readme? I think that can be ignored.

but yeah, if it will be used as _base_ for other PRs coverage comparison, then we gotta have it :+1: